resnet.py

```diff
@@ -0,0 +1,94 @@
+import json
+from pathlib import Path
+from typing import Any, Optional, Tuple
+
+import torch
+import torchvision.models as models
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.nn import BCEWithLogitsLoss, Linear
+from torch.optim import AdamW
+from torch.optim.lr_scheduler import CosineAnnealingLR
+from torchio.transforms import RescaleIntensity
+
+from config import Config
+
+
+class LitResnet(LightningModule):
+    def __init__(self, out_features: int, config: Config, resnet_version: int = 18) -> None:
+        super().__init__()
+        resnets = {
+            18: models.resnet18,
+            34: models.resnet34,
+            50: models.resnet50,
+            101: models.resnet101,
+            152: models.resnet152,
+        }
+        self.out_features = out_features
+        self.bce = BCEWithLogitsLoss()
+        self.resnet_version = resnet_version
+        self.resnet_model = resnets[resnet_version](pretrained=True)
+        linear_size = list(self.resnet_model.children())[-1].in_features
+        self.resnet_model.fc = Linear(linear_size, out_features)
+        self.config = config
+
+    def save_configs(self, log_dir: Path):
+        log_dir = Path(log_dir) / "config.json"
+        self.config.__dict__["resnet_version"] = self.resnet_version
+        json.dump(self.config.__dict__, open(log_dir, "w"))
+        # DataFrame(self.config).to_json(log_dir_version)
+
+    def forward(self, x, *args, **kwargs) -> Any:
+        x = self.resnet_model(x)
+        return x
```

Comment:
I know typing kind of sucks for Lightning model core loop functions, but surely we can do better than this:
```suggestion
    def forward(self, x: Tensor, *args: Mapping, **kwargs: Mapping) -> Tensor:
        x = self.resnet_model(x)
        return x
```

Reply:
I thought I would do that at the end lol